src/main/java/org/jboss/modules/xml/ModuleXmlParser.java

```diff
@@ -902,20 +911,31 @@ private static void parseNativeArtifact(final MavenResolver mavenResolver, final
             throw missingAttributes(reader, required);
         }
 
+        final SystemPropertyConditionBuilder conditionBuilder = new SystemPropertyConditionBuilder();
+        final Set<String> encountered = new HashSet<>();
         int eventType;
         for (;;) {
             eventType = reader.nextTag();
             switch (eventType) {
                 case END_TAG: {
                     try {
-                        createMavenNativeArtifactLoader(mavenResolver, name, reader, specBuilder);
+                        if(conditionBuilder.resolve()) {
```

Comment:
Not a problem @stuartwdouglas but I noticed you're using strange indentation for conditionals (missing space between conditional type and first left bracket). Would be nice to have it synced with others.